File: ccc/2013/j2.py
#https://dmoj.ca/problem/ccc13j2
#CCC '13 J2 - Rotating letters

ab = input()
a = 0
b = 0

ablist = list(ab)

for x in ablist:
    if x == "I":
        a += 1
    elif x == "O":
        a += 1
    elif x == "S":
        a += 1
    elif x == "H":
        a += 1
    elif x == "Z":
        a += 1
    elif x == "X":
        a += 1
    elif x == "N":
        a += 1
    else:
        b += 1

if b == 0:
    print("YES")
else:
    print("NO")

File: ccc/2010/j2.java
//https://dmoj.ca/problem/ccc10j2
//CCC '10 J2 - Up and Down

import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args){
		Scanner input = new Scanner(System.in);
        int a = input.nextInt();
        int b = input.nextInt();
        int c = input.nextInt();
        int d = input.nextInt();
        int s = input.nextInt();
        int nikkySteps = 0; 
        int nikkysDistance = 0;
        int next = a;
        int sign = 1;
        while(nikkySteps + next < s){
            nikkySteps = nikkySteps + next;
            nikkysDistance = nikkysDistance + sign * next;
            if(sign == 1){
                next = b;
            }
            else{
                next = a;
            }
            sign = sign * -1;
        }
        nikkysDistance = nikkysDistance + sign * (s - nikkySteps);
            int byronSteps = 0; 
            int byronsDistance = 0;
            next = c;
            sign = 1;
            while(byronSteps + next < s){
                byronSteps = byronSteps + next;
                byronsDistance = byronsDistance + sign * next;
                if(sign == 1){
                    next = d;
                }
                else{
                    next = c;
                }
                sign = sign * -1;
            }
            byronsDistance = byronsDistance + sign * (s - byronSteps);
        if(nikkysDistance > byronsDistance)System.out.println("Nikky");
            else if(byronsDistance > nikkysDistance)System.out.println("Byron");
            else System.out.println("Tied");
        input.close();
	}
}
